### PR TITLE
use requests instead of urllib

### PR DESCRIPTION
**Pull request by jnozsc**
as https://github.com/john-kurkowski/tldextract/pull/88#discussion_r52067412 mentioned

### CODE REVIEW

**Comment by jnozsc** [comment]
(1) `requests` can handle gzip by default http://docs.python-requests.org/en/latest/community/faq/#encoded-data
(2) `requests` can not handle `file://` so I have to add a `requests-file`

**Comment by john-kurkowski** [comment]
Oh wow! I was idly wishing in [that comment](https://github.com/john-kurkowski/tldextract/pull/88#discussion_r52067412). This is pretty nice.

For tldextract 1.x, I'd like to stick to no external dependencies. Mind if I wait to merge this until 2.x?

File: requirements.txt [hunk 1]
```diff
@@ -5,4 +5,6 @@ pytest==2.7.3
 pytest-gitignore==1.3
 pytest-mock==0.9.0
 pytest-pylint==0.4.0
+requests>=2.9.1
```
Comment (john-kurkowski) [inline requirements.txt]: Should this be `>= 2.0, < 3`? Maybe I've been in Node.js too long.

Comment (john-kurkowski) [inline requirements.txt]: Does this need to be in setup.py too?

**Comment by jnozsc** [comment]
no problem, let's keep this open until 2.x

File: requirements.txt (hunk 1 above)
Comment (john-kurkowski) [inline requirements.txt]: Thanks!

**Comment by mauricioabreu** [comment]
Good job @jnozsc 
urllib has some problems that the vendored urlblib in requests does not have.

**Comment by mauricioabreu** [comment]
This PR can be closed, right? It is already merged in 2.X branch :)

**Comment by john-kurkowski** [comment]
I was gonna let it autoclose when it went to `master`, but sure, closed via 6e4f9cf4cff2254bbd98ed4c9adbb744f4b5e564. Thanks @jnozsc!

**Comment by john-kurkowski** [comment]
Never mind. Keeping this open until autoclose on `master`. If I close it manually, the PR is marked "Closed," not "Merged," which is misleading.

**Comment by TylerLubeck** [comment]
Hi all,

Is pinning `requests>2.9.1,<3` necessary? We have a few packages that rely on both tldextract and requests 2.2 (long story), and we're getting version conflicts because of it.

I ran the test suite with requests 2.2.1, requests 2.8.1, and requests 2.9.1 and they all pass.

**Comment by john-kurkowski** [comment]
The exact version range was somewhat arbitrary. I think it can be more flexible. Thanks for #98! Let's discuss there.

**Pull request merged**